quinn/src/send_stream.rs

```diff
@@ -245,6 +252,32 @@ impl SendStream {
     }
 }
 
+/// Check if a send stream is stopped.
+///
+/// Returns `Some` if the stream is stopped or the connection is closed.
+/// Returns `None` if the stream is not stopped.
+fn send_stream_stopped(
+    conn: &mut State,
+    stream: StreamId,
+    is_0rtt: bool,
+) -> Option<Result<Option<VarInt>, StoppedError>> {
+    if is_0rtt && conn.check_0rtt().is_err() {
+        Some(Err(StoppedError::ZeroRttRejected))
+    } else {
+        match conn.inner.send_stream(stream).stopped() {
+            Err(ClosedStream { .. }) => Some(Ok(None)),
+            Ok(Some(error_code)) => Some(Ok(Some(error_code))),
+            Ok(None) => {
+                if let Some(error) = &conn.error {
```

Comment:
I think this could be a `map()` call? `conn.error.cloned().map(Into::into)` maybe?

Reply:
It could, again I preferred this form because for me it reads much clearer this way with the quite complex return type. For me, the `if let` is very straightforward to read, where as the `cloned().map(Into::into)` I have to take two or three looks to be sure what exactly is going on. But again, your call ofc.

Reply:
Ah well, clippy complains about my manual if-let, so changing it to the map.